stablehlo/tests/interpret_case.mlir

```diff
@@ -0,0 +1,15 @@
+// RUN: stablehlo-interpreter --interpret -split-input-file %s
+
+func.func @case() {
+  %index = stablehlo.constant dense<-1> : tensor<i32>
```

Comment:
Sounds like we need three tests here: 1) less than zero, 2) in bounds, 3) greater or equal than the number of branches.